### PR TITLE
[AzureMonitorDistro] Update Readme with advanced configuration

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 1]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+builder.Services.Configure<SqlClientInstrumentationOptions>(options =>
+{
+    options.SetDbStatementForStoredProcedure = false;
+});
+```
+
+#### Customizing Sampling Percentage
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor()
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:132]: not aligned properly?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 2]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
 
 Note that the `Credential` property is optional. If it is not set, Azure Monitor Distro will use the Instrumentation Key from the Connection String to send data to Azure Monitor.
 
+### Advanced configuration
+
+The Azure Monitor Distro includes .NET OpenTelemetry instrumentation for [ASP.NET Core](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNetCore/), [HttpClient](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Http/), and [SQLClient](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.SqlClient). However, you can customize the instrumentation included or use additional instrumentation on your own using the OpenTelemetry API. Here are some examples of how to customize the instrumentation:
+
+#### Customizing ASP.NET Core Instrumentation Options
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:89]: consider linking to the instrumentation library readme here to help users find the full options available

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 3]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+builder.Services.AddOpenTelemetry().UseAzureMonitor()
+                                   .WithTracing(b =>
+                                   {
+                                       b.SetSampler(new ApplicationInsightsSampler(0.9F));
+                                   });
+```
+
+(or)
+
+```
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:141]:
```suggestion
```C#
```

Needs the title to display correct formatting.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 4]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+builder.Services.ConfigureOpenTelemetryTracerProvider((sp, builder) => builder.SetSampler(new ApplicationInsightsSampler(0.9F)));
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:142]: All of the other examples are multi-line. 
For consistency, consider formatting this the same.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:132]: Looks like this is aligned with the ".UseAzureMonitor().
I don't have a strong opinion on the alignment, as long as we're consistent. :)

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 5]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
 
 Note that the `Credential` property is optional. If it is not set, Azure Monitor Distro will use the Instrumentation Key from the Connection String to send data to Azure Monitor.
 
+### Advanced configuration
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:85]: I recommend adding a section just before this header to document what is included in the distro. Which sampler, which instrumentations, which exporters, resource detectors etc. Without knowing what is included, there advanced configuration section lacks context.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 6]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor()
+                                   .WithTracing(b =>
+                                   {
+                                       b.SetSampler(new ApplicationInsightsSampler(0.9F));
+                                   });
+```
+
+(or)
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:139]: do we need to show both?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 7]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+
+```
+builder.Services.ConfigureOpenTelemetryTracerProvider((sp, builder) => builder.SetSampler(new ApplicationInsightsSampler(0.9F)));
+```
+
+#### Adding Custom ActivitySources to Traces
+
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:146]: describe here (or earlier) what sources are subscribed by default.

Comment (cijothomas) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:146]: for example, say the distro enables the sources from the instrumentaion library A,B,C and it also includes the source "Azure.*".

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 8]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+                                   });
+```
+
+(or)
+
+```
+builder.Services.ConfigureOpenTelemetryTracerProvider((sp, builder) => builder.SetSampler(new ApplicationInsightsSampler(0.9F)));
+```
+
+#### Adding Custom ActivitySources to Traces
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:145]:
```suggestion
#### Adding Custom ActivitySource to Traces
```

No preference, but please ensure that it is consistent.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 9]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+                                       b.AddMeter("MyCompany.MyProduct.MyLibrary");
+                                   });
+```
+
+#### Adding Additional Instrumentation
+
+If you need to instrument a library or framework that isn't included in the Azure Monitor Distro, you can add additional instrumentation using the OpenTelemetry Instrumentation packages. For example, to add instrumentation for gRPC clients, you can add the [OpenTelemetry.Instrumentation.GrpcNetClient](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.GrpcNetClient/) package and use the following code:
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor()
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:170]: it'd be good to mention that irrespective of the number of times this is called, there is single tracer/meter provider only, and every config is simply being applied to that single one. And call out that if additional provider is needed, hand-construction is needed.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 10]
```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+                                   .WithMetrics(b =>
+                                   {
+                                       b.AddMeter("MyCompany.MyProduct.MyLibrary");
+                                   });
+```
+
+#### Adding Additional Instrumentation
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:165]: lets also show.
1. Add custom Resource
2. Add additional Exporter.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md (hunk 2 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:89]: We need to link this to a GitHub page, which might change. Better we get into this details when we publish docs in docs.microsoft.com.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md (hunk 6 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:139]: Changing all of the samples to use `builder.Services.ConfigureOpenTelemetryTracerProvider` pattern.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 11]
```diff
@@ -82,6 +85,125 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
 
 Note that the `Credential` property is optional. If it is not set, Azure Monitor Distro will use the Instrumentation Key from the Connection String to send data to Azure Monitor.
 
+### What is Included in the Distro
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:88]: This write up is good, but it's in the "Getting Started" section.
I think this belongs earlier in the document, I suggest before line 5.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 12]
```diff
@@ -65,6 +65,9 @@ var app = builder.Build();
 
 Note that in the examples above, `UseAzureMonitor` is added to the `IServiceCollection` in the `Program.cs` file. You can also add it in the `ConfigureServices` method of your `Startup.cs` file.
 
+> **Note**
+  > Multiple calls to `AddOpenTelemetry.UseAzureMonitor()` will **NOT** result in multiple providers. Only a single `TracerProvider` and/or `MeterProvider` will be created in the target `IServiceCollection`. To establish multiple providers use the `Sdk.CreateTracerProviderBuilder()` and/or `Sdk.CreateMeterProviderBuilder()` methods with the [Azure Monitor Exporter](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter) instead of using Azure Monitor Distro.
```
[the remark this answers is not in the document]
Comment (reyang) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:69]: > Only a single `TracerProvider` and/or `MeterProvider` will be created

What about logger?

[the remark this answers is not in the document]
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:69]: OpenTelemetry Hosting API follows this pattern. Calling `WithTracing` multiple times does not throw. As we are part of it, better to align with its principles.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 13]
```diff
@@ -82,6 +85,125 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
 
 Note that the `Credential` property is optional. If it is not set, Azure Monitor Distro will use the Instrumentation Key from the Connection String to send data to Azure Monitor.
 
+### What is Included in the Distro
+
+The Azure Monitor Distro is a distribution of the .NET OpenTelemetry SDK and related instrumentation libraries. It includes the following components:
+
+* Traces
+  * [ASP.NET Core Instrumentation Library](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNetCore/) provides automatic tracing for incoming HTTP requests to ASP.NET Core applications.
+  * [Http Client Instrumentation Library](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Http/) provides automatic tracing for outgoing HTTP requests made using [System.Net.Http.HttpClient](https://docs.microsoft.com/dotnet/api/system.net.http.httpclient) and [System.Net.HttpWebRequest](https://docs.microsoft.com/dotnet/api/system.net.httpwebrequest).
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:94]:
```suggestion
  * [HTTP Client Instrumentation Library](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Http/) provides automatic tracing for outgoing HTTP requests made using [System.Net.Http.HttpClient](https://docs.microsoft.com/dotnet/api/system.net.http.httpclient) and [System.Net.HttpWebRequest](https://docs.microsoft.com/dotnet/api/system.net.httpwebrequest).
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:69]: > What about logger?

It is applicable for logger too, will add to this message.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 14]
```diff
@@ -82,6 +103,107 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+builder.Services.AddOpenTelemetry().UseAzureMonitor();
+builder.Services.Configure<SqlClientInstrumentationOptions>(options =>
+{
+    options.SetDbStatementForStoredProcedure = false;
+});
+```
+
+#### Customizing Sampling Percentage
+
+``` C#
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:152]: the doc is missing what is the defaut, if this section is followed by the user.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 15]
```diff
@@ -82,6 +103,107 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+builder.Services.ConfigureOpenTelemetryTracerProvider((sp, builder) => builder.AddGrpcClientInstrumentation());
+```
+
+#### Adding Another Exporter
+
+Azure Monitor Distro uses the Azure Monitor exporter to send data to Application Insights. However, if you need to send data to other services, including Application Insights, you can add another exporter. For example, to add the Prometheus exporter, you can install the [OpenTelemetry.Exporter.Prometheus.AspNetCore](https://www.nuget.org/packages/OpenTelemetry.Exporter.Prometheus.AspNetCore) package and use the following code:
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor();
+builder.Services.ConfigureOpenTelemetryMeterProvider((sp, builder) => builder.AddPrometheusExporter());
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:186]: this is probably a poor example, as prometheus require further setup! 
I suggest to show Console OR OTLP.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 16]
```diff
@@ -82,6 +103,107 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+```
+
+#### Adding Custom Resource
+
+To customize the resource, use the following code.
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:191]: still missing the default.

**Review by cijothomas — APPROVED** [review]
Looks good overall. Left some comments, which are good to have in follow ups.
(The main suggestion is to include what is the default settings for each provider in this doc itself, before we talk about how to change defaults)

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md (hunk 16 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:191]: There are no defaults, yet to be added. We could update this section when we add resource defaults.

**Review by reyang — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 17]
```diff
@@ -82,6 +103,109 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor();
+builder.Services.Configure<HttpClientInstrumentationOptions>(options =>
+{
+    options.RecordException = true;
+    options.FilterHttpRequestMessage = (httpRequestMessage) =>
+    {
+        // only collect telemetry about HTTP GET requests
+        return httpRequestMessage.Method.Equals("GET");
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:135]: Is this the idiomatic way? (or https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httpmethods.isget?view=aspnetcore-7.0)

Comment (reyang) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:135]: Found a similar discussion https://github.com/IdentityServer/IdentityServer4/issues/2098

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md [hunk 18]
```diff
@@ -82,6 +103,109 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
...
+    };
+});
+```
+
+#### Customizing SQLClient Instrumentation Options
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md:140]:
```suggestion
#### Customizing SqlClientInstrumentationOptions
```

**Review by reyang — APPROVED** [review]

**Pull request merged**